torchsynth/module.py

```diff
@@ -377,36 +344,38 @@ class TorchVCO(TorchSynthModule1D):
 
     Parameters
     ----------
-    midi_f0 (T)     :   pitch value in 'midi' (69 = 440Hz).
-    mod_depth (T)   :   depth of the pitch modulation in semitones.
+    synthglobals: TorchSynthGlobals        : global args, see TorchSynthModule
     phase (optional, T)       :   initial phase values
-    **kwargs        : keyword args, see TorchSynthModule1D
     """
 
-    def __init__(self, midi_f0: T, mod_depth: T, phase: Optional[T] = None, **kwargs):
-        TorchSynthModule1D.__init__(self, batch_size=midi_f0.shape[0], **kwargs)
-        self.add_parameters(
-            [
-                ModuleParameter(
-                    value=midi_f0,
-                    parameter_name="pitch",
-                    parameter_range=ModuleParameterRange(0.0, 127.0),
-                ),
-                ModuleParameter(
-                    value=mod_depth,
-                    parameter_name="mod_depth",
-                    parameter_range=ModuleParameterRange(0.0, 127.0),
-                ),
-            ]
-        )
+    parameter_ranges: List[ModuleParameterRange] = [
+        ModuleParameterRange(
+            0.0, 127.0, name="midi_f0", description="pitch value in 'midi' (69 = 440Hz)"
+        ),
+        ModuleParameterRange(
+            0.0,
+            127.0,
+            name="mod_depth",
+            description="depth of the pitch modulation in semitones",
+        ),
+        # TODO: I think phase should go in here??
```

Comment:
I don't think we need phase as a ranged parameter. It's sort of a weird one -- it's important to keep track of to avoid clicks on consecutive calls to forward, but varying it like a regular synth parameter isn't going to do that much perceptually. I think pretty much always we are just going to want it to be initialized to zero on construction. Interested in @maxsolomonhenry thoughts.

Reply:
oki then we leave it as is